sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishingEvents.md

```diff
@@ -0,0 +1,363 @@
+
+The `EventDataBatch` exists to provide a deterministic and accurate means to measure the size of a message sent to the service, minimizing the chance that a publishing operation will fail.  Because the batch works in cooperation with the service, it has an understanding of the maximum size and has the ability to measure the exact size of an event when serialized for publishing.  For the majority of scenarios, we recommend using the `EventDataBatch` to ensure that your application does not attempt to publish a set of events larger than the Event Hubs service allows.  The majority of examples in this sample will demonstrate a batched approach.
+
+To create an `EventDataBatch`, the `EventProducerClient` must be used, as the size limit is queried from the Event Hubs service the first time that a batch is created.  After the size has been queried once, batch creation will not incur the cost of a service request.   The `EventDataBatch` follows a `TryAdd` pattern; if the call returns `true` then the event was accepted into the batch.  If not, then the event was unable to fit.  To avoid accidentally losing events, it is recommended to check the return value when adding events.
+
+```C# Snippet:EventHubs_Sample04_EventBatch
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+try
+{
+    using var eventBatch = await producer.CreateBatchAsync();
+
+    var eventBody = new BinaryData("This is an event body");
+    var eventData = new EventData(eventBody);
+
+    if (!eventBatch.TryAdd(eventData))
+    {
+        throw new Exception($"The event could not be added.");
+    }
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+The `EventDataBatch` is scoped to a single publish operation.  Once that operation is complete, a new batch should be created for any additional events to be published.  The batch is responsible for unmanaged resources; it is recommended that you `Dispose` the batch after it has been published.
+
+## Publishing events with automatic partition assignment
+
+Allowing automatic assignment to partitions is recommended when publishing needs to be highly available and shouldn't fail if a single partition is experiencing trouble.  Automatic assignment also helps to ensure that event data is evenly distributed among all available partitions, which helps to ensure throughput when publishing and reading data.
+
+When the producer publishes the event, it will receive an acknowledgment from the Event Hubs service; so long as there is no exception thrown by this call, the service is now responsible for delivery.  Your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
```

Comment:
```suggestion
When the producer publishes the event, it will receive an acknowledgment from the Event Hubs service; so long as there is no exception thrown by this call, the service assumes responsibility for delivery.  Your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
```

Reply:
Much, much better....  thank you!